### DISCUSSION TITLE
Поезд - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#7889] [Docs]
# Поезд - Ultralytics YOLOv8 Docs

Пошаговое руководство по обучению моделей YOLOv8 с помощью Ultralytics YOLO  , включая примеры обучения на одном GPU и нескольких GPU.

https://docs.ultralytics.com/ru/modes/train/

**Comment by EVNN304** [comment]
Hello. Can you please tell me what and where to change in the neural network settings in order to learn from black and white images. Thanks in advance for your feedback

**Reply by pderrenger** [reply]
@Egorvasechko hello! To train a YOLOv8 model on black and white images, you'll need to adjust the input channels of the model to accept single-channel images instead of the default three-channel color images. Here's a concise guide:

1. **Prepare Your Dataset**: Ensure your dataset images are converted to grayscale. You can use image processing tools like OpenCV to convert color images to black and white.

2. **Modify Configuration**:
   - Locate the model configuration file (YAML file), which defines the model architecture.
   - In the YAML file, find the input channels parameter (usually denoted as `nc` or `channels`) and change it from `3` to `1`.

3. **Update Data Loading**:
   - Modify the data loading code to read images as grayscale. In PyTorch, this involves setting the `color_mode` to `grayscale` when loading the images.

4. **Train the Model**: Train your model with the updated configuration and grayscale images.

Here's an example of how you might adjust the data loading in Python:

```python
from PIL import Image

# Load the image in grayscale
image = Image.open('path/to/image.jpg').convert('L')
```

Remember to apply these changes consistently across your entire dataset and training pipeline. If you encounter any issues or have further questions, feel free to reach out! 🚀
